internal/controller/networkpolicy/np_controller.go

```diff
@@ -70,11 +107,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result
 }
 
 func (r *Reconciler) reconcile(ctx context.Context, clh *helper.Client, desired *flowslatest.FlowCollector) error {
+	l := log.FromContext(ctx)
+
 	cni, err := r.mgr.ClusterInfo.GetCNI()
 	if err != nil {
 		return err
 	}
-	npName, desiredNp := buildMainNetworkPolicy(desired, r.mgr, cni)
+
+	// Get API server endpoint IPs for network policy
+	var apiServerIPs []string
+	if r.mgr.ClusterInfo.IsOpenShift() {
+		apiServerIPs, err = GetAPIServerEndpointIPs(ctx, r.Client)
+		if err != nil {
+			l.Error(err, "Failed to get API server endpoint IPs, network policy will allow all IPs on port 6443")
```

Comment:
I think I'm rather for not creating the egress rule at all, than creating a less secure one. Ie. I'm rather for having it breaking the connectivity, forcing the user to troubleshoot the issue, rather than opening all egress on 6443 somewhat silently